### ISSUE TITLE
GUI update

### ISSUE THREAD

**Issue by jonasoreland** [jonasoreland/runnerup#295]
Maybe it's time to update the GUI.
What do you think?

**Comment by bagage** [comment]
Yeeeah :) Do you know any good UI designer (not that you aren't one but... eh, we're developers right? :)).

**Comment by paradix** [comment]
Great idea indeed, but ehh that's a lot of work and good UX experience would be helpful.

**Comment by cmmata** [comment]
I discovered this app a few weeks ago, and when I saw Play's screenshots I thought it was an abandoned project. Then I saw last update date and I realized that it's still alive and active. So I think it's a great idea to update the GUI :) Not everybody takes a look at update dates, so maybe other people simply leaves

**Comment by ixeft** [comment]
Hello,

I started some time ago to think about a new UI for RunnerUp.
It is very raw and far from finished and I'm not an UI designer, but maybe it could be a start.

https://free-ixeft42.proto.io/player/?id=04e3b4a5-a562-44f7-833f-a16f9cf2d335

What do you think about that ? (I won't take it bad if you just say that it is ugly.)

I got plenty of Ideas for others views but not had the time to finish it.

**Comment by nebmo** [comment]
Do you have screens or do I have to sign up?

**Comment by nebmo** [comment]
I signed up but can't access yours. Have to enter login details

**Comment by ixeft** [comment]
Hello,
I was persuated that it was possible to have a public access to the sketch but it's not ...
I'm posting screens...

**Comment by ixeft** [comment]
![home](https://cloud.githubusercontent.com/assets/422722/9418500/daa02342-4854-11e5-84ae-8601fbef6711.png)
![hrz](https://cloud.githubusercontent.com/assets/422722/9418504/daa2b698-4854-11e5-96e1-ae52cc4bb81e.png)
![interval](https://cloud.githubusercontent.com/assets/422722/9418502/daa16d42-4854-11e5-836e-d646f0b02d0b.png)
![menu](https://cloud.githubusercontent.com/assets/422722/9418501/daa14628-4854-11e5-8cd1-a6f0fa61ed86.png)
![time](https://cloud.githubusercontent.com/assets/422722/9418503/daa1c350-4854-11e5-94ee-52159ab6b08d.png)

**Comment by ixeft** [comment]
![time-small](https://cloud.githubusercontent.com/assets/422722/9418610/81fc129a-4855-11e5-9aa7-ba4bf7243445.png)
![home-small](https://cloud.githubusercontent.com/assets/422722/9418612/880dc200-4855-11e5-8b88-767310fcde2f.png)
![hrz-small](https://cloud.githubusercontent.com/assets/422722/9418613/8824d8aa-4855-11e5-8c02-d83ab090ce8e.png)
![interval-small](https://cloud.githubusercontent.com/assets/422722/9418614/8826d45c-4855-11e5-988c-9ffd66eacf22.png)
![menu-small](https://cloud.githubusercontent.com/assets/422722/9418615/882e82ba-4855-11e5-897c-0b6451979cf1.png)

**Comment by daverix** [comment]
I would rather have some Material Design in RunnerUp. Here is a mock of how it could look:

![runnerup_main](https://cloud.githubusercontent.com/assets/55992/9439375/9d304d70-4a66-11e5-9236-eeaa49306994.png)

The idea is that you might use the app more often for checking your feed than to actually run so that's why it's shown first. Then at the bottom you have quick access to configure and start a run. Pressing on the GPS square would let you chose if you want to use an external gps like the one in some Wear watches or if you want to use the GPS on your phone.

Configuration will show a view similar to what is now where you can choose between the json files from garmin, the interval config or just plain basic with a target pace.

**Comment by ixeft** [comment]
Sure it's a good idea to use material design !

I like the idea to have the four controls buttons to get quick access to some settings. 
(For me, Audio cue should be there too)
- for GPS : it can also be a screen giving more informations about GPS fix

In my opinion, a general stats view (total run this month, average speed, etc.) is more used that feeds.
but, I also like the idea to keep the configuration screen as homepage : the sooner you configure your run, the quicker you start running, and that's what matter, no ? :p

**Comment by daverix** [comment]
Yeah I used proto.io, a trial, just searched for material design mock on google, then took a screenshot :)

**Comment by ixeft** [comment]
Haha, so I'm just a bad at it ^^

**Comment by jonasoreland** [comment]
I  like both sort of, they look much more modern that what we have today.

What I like about today is,
1) 1 click for starting basic run
2) 2 click for starting interval/advanced same as last

re daverix: I wonder if people "on average" does look at feed frequently, I
know I don't...every know and then, but not frequently.
I mainly chose what kind of run to run...and then do that.
I sometime look at history, but most often I do that online instead.

/Jonas

**Comment by darander** [comment]
I'll have to agree with Jonas here. When I'm starting RunnerUp, it's almost always to record a run. This having a 1 or 2 click approach to start a new session (just as it's today and as Jonas described it above) is something that's really appealing. 

If someone wants the feed as the first thing to site up, that could maybe be done as an option. But the default should really be today's behaviour.

**Comment by RunningCl0ud** [comment]
Yepp. I really like to hit the button and hit the road! I have wi-fi for checking on my friends on different obscured sites and often do that with the computer. Also I like the gui for being kind of clean.

**Comment by tbdal** [comment]
what about using the Materialdrawer framework?
there are already many options included that may help creating a nice ui experience
http://mikepenz.github.io/MaterialDrawer/

**Comment by cmmata** [comment]
What's the status of the GUI update? If nobody is working on it, I could help with that. I recently finished an Android Developer course and I want to practice.

**Comment by jonasoreland** [comment]
patches always welcome!

**Comment by cmmata** [comment]
Okay, I'll take a look at this, based on @daverix and @ixeft ideas and the last material design principles. I'll add snapshots to pull requests so anybody can see the design and discuss changes.

**Comment by gerhardol** [comment]
Material Design with App Bar (Action bar), side navigation and a floating + (one click click start) should be enough.
The current tabs should be removed. The main tasks like Basic workout, Interval, feed, settings should be separate items in the side nav bar. Each individual History item should be one scrollable part, no separate tab (possibly bottom tab to jump).

RunnerUp will then look the same as most other "competitors". This is maybe not better in itself, but it is a standard interface that users are familiar with.
Other tweaks to the structure should be done after the structure is changed. (Should you be allowed to check history etc while recording)

Steps:
Decide what minimal version that should be supported in the new UI (older versions still have the old UI). Various workarounds are needed for devices older than Android 5.0
Make all Androids Activities AppCompat. I have started this on a private branch, this is done on top of unmerged #524.
Add App Bar and side nav bar. The app bar is in this step really just using space.
Add current main activities to the side nav bar.
Split the current "top" activity to separate activities so they can be added separately.
(Use a fragment based UI.) 
(Remove tab interface for History).

**Comment by jonasoreland** [comment]
SGTM
👍

**Comment by cmmata** [comment]
I think this should be according to the "RunnerUp Classic" and "RunnerUp Classic Lite" versions. As @gerhardol said in #586, devices older than 4.0.1 get the classic app. So the new UI should focus in versions > 4.0.1, right? I'll take a look if there's something too much difficult or impossible to port below 5.0, but I think AppCompat should work.

The [Android developer console chart](https://developer.android.com/about/dashboards/index.html) shows that there's a 0.7% using 4.0.3-4.0.4. Should I focus on 4.1.x (Jelly Bean)? Or do you recommend to directly get the minimum of 4.4 (17% usage)? There's a 9% usage bellow 4.4.

**Comment by gerhardol** [comment]
See #404 for the RU distribution.
RunnerUp users are more on older devices - what you drag along on a run. RU supports many older HR protocols, now most phones support BLE so that support is less important. Also, the current version does not have features as Cadence (in latest beta though) and has an outdated look (no much difference in using, but may not appear to new users). 
Nevertheless, developing for never versions instead of doing nothing is not taking anything away from older versions.

I propose 4.4 or 5.0 as minimal versions.

**Comment by dapgo** [comment]
Agreed. RU is the perfect type of app to use on our old devices. In fact new devices are used to be too big and heavy to wear on the pocket or arm. Keep it backward compatible, please :)

**Comment by gerhardol** [comment]
Discussions in #586 
The "Classic" will stay as-is with new development targetting 5.0 and later

**Comment by davidedelvento** [comment]
I used to think like @dapgo -- however after having used RU for a while on a old (very lightweight) Samsung running Android v2.3 or so, I changed my mind. Sure, the phone is small and light, but is too slow to be useful. Moreover, **many** android activities or fragments ("windows" in PC parlance) are too big for the screen and cannot be scrolled, so there are many things that I cannot press, and a few that I can press by being **extremely** careful to hit the only visible pixel of the button.

@gerhardol I think with the great issue cleanup you are doing (many thanks!!!) you can close this one. We know @cmmata is working on this, and wait for PR from him (or make ours if he doesn't in several weeks)

**Comment by cmmata** [comment]
I'm taking a look at Material Design principles while waiting for #524, so I can start with a cleaner version of RU. If it takes too long, I'll start anyway and merge later.

Anyway, why closing this issue? It can be useful to keep it to talk about the new GUI.

**Comment by davidedelvento** [comment]
@cmmata I think closing this issue make sense. If you are working on it, why don't you create a corresponding issue on your fork as described in #586 (including the "linking" between that one and this one)? I think having not too many issues open here is better, and the current 89 issues is too many.

**Comment by cmmata** [comment]
👍🏻 okay, I'll do it that way

**Comment by gerhardol** [comment]
I suggest this is kept open.
I have assigned the milestone for it (as well as a project, may remove that.)

**Comment by gerhardol** [comment]
I have pushed some changes I used when playing around with the GUI some months ago. This primarily uses AppCompatActivity and removes some depreciated functionality (like TabActivity).
https://github.com/gerhardol/runnerup/tree/feature/appcompat-changes

Based on master_ng, but could be used in master (needed to get menu overflow in Lollipop and later).
Mostly untested

**Comment by 12people** [comment]
I made some mockups that bring the UI up-to-date to the material design guidelines: https://www.figma.com/file/SlPQaT56OztqaLmFZAeM0y/RunnerUp . They're only visual/layout changes and don't add, move, or remove any functionality—that should make it easier to implement and easier to adapt to for current users.

Not every screen is there yet.

Feel free to duplicate the mockup and work on it:
![Duplicate in Figma](https://storage.jumpshare.com/preview/richBvyOu3JibmTn3yc_ZOaYj1EZ_r3B9O9QtWZfEIJR5LUhIq9IdeVe34YVPGmA7GguoeZHQOblPVIAlD56Jt0Iq-_ZMIwlJNqsu6s4bO0F1kR3dMUjedqC16uBUu85)

As for a side navbar—please don't add one. There are only 4 navigation options and Google's guidelines recommend [bottom navigation](https://material.io/guidelines/components/bottom-navigation.html#bottom-navigation-usage) for "Three to five top-level destinations of similar importance".

**Comment by 12people** [comment]
I also added "V2" and "V3" screens to show how I think the UI might evolve—each version is more complex to implement than the one before, animations are posted as comments.

**Comment by 12people** [comment]
I've updated the mockups to reflect what I'd like to accomplish in the end: https://www.figma.com/file/SlPQaT56OztqaLmFZAeM0y/RunnerUp . Functionally, it's the same, but it's made to look nicer and be more usable.

Please take a look and comment.

**Comment by guglovich** [comment]
I am making a UX design change in Figma. Rearrange elements and new usability, everything has to be in a couple of clicks and the user doesn't have to think. I won't touch the UI for now, I'll leave the old palette for now, but in the future I might do a Neomorphism style, for example.

**Comment by gerhardol** [comment]
Especially the buttons rearrangement is an improvement. (Some options can be pruned like to only have Lap/Pause while running).
I believe that @12people did a great job with the changes, but most tunings will be an improvement.
